test/test_active.py

```diff
@@ -0,0 +1,277 @@
+            self.assertTrue(test_passed, msg=msg)
+            self.assertTrue(hasattr(encrypted_tensor, "_mac"), "_mac attribute failed to propagate")
+
+        if dst is None or dst == self.rank:
+            tensor = encrypted_tensor._tensor.reveal(dst=dst)
+            alpha = MPCTensor.alpha.reveal(dst=dst)
+            mac = encrypted_tensor._mac.reveal(dst=dst)
+            self.assertTrue((alpha * tensor == mac).all(), msg=f"{msg} - Mac Arithmetic Mismatch")
+        elif dst is not None:
+            self.assertIsNone(encrypted_tensor._tensor.reveal(dst=dst))
+            self.assertIsNone(MPCTensor.alpha.reveal(dst=dst))
+            self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst))
+
+    def test_encrypt_decrypt(self):
+        """
+        Tests tensor encryption and decryption for both positive
+        and negative values.
+        """
+        sizes = [
+            (),
+            (1,),
+            (5,),
+            (1, 1),
+            (1, 5),
+            (5, 1),
+            (5, 5),
+            (1, 5, 5),
+            (5, 1, 5),
+            (5, 5, 1),
+            (5, 5, 5),
+            (1, 3, 32, 32),
+            (5, 3, 32, 32),
+        ]
+        for size in sizes:
+
+            # encryption and decryption without source:
+            reference = self._get_random_test_tensor(size=size, is_float=True)
+            encrypted_tensor = MPCTensor(reference)
+            self._check(encrypted_tensor, reference, "en/decryption failed")
+            for dst in range(self.world_size):
+                self._check(
+                    encrypted_tensor, reference, "en/decryption failed", dst=dst
+                )
+
+            # encryption and decryption with source:
+            for src in range(self.world_size):
+                input_tensor = reference if src == self.rank else []
+                encrypted_tensor = MPCTensor(input_tensor, src=src, broadcast_size=True)
+                for dst in range(self.world_size):
+                    self._check(
+                        encrypted_tensor,
+                        reference,
+                        "en/decryption with broadcast_size failed",
+                        dst=dst,
+                    )
+
+            # test creation via new() function:
+            encrypted_tensor2 = encrypted_tensor.new(reference)
+            self.assertIsInstance(
+                encrypted_tensor2, MPCTensor, "new() returns incorrect type"
+            )
+            self._check(encrypted_tensor2, reference, "en/decryption failed")
+
+        # MPCTensors cannot be initialized with None:
+        with self.assertRaises(ValueError):
+            _ = MPCTensor(None)
+
+    def test_arithmetic(self):
+        """Tests arithmetic functions on encrypted tensor."""
+        arithmetic_functions = ["add", "add_", "sub", "sub_", "mul", "mul_"]
+        for func in arithmetic_functions:
+            for tensor_type in [lambda x: x, MPCTensor]:
```

Comment:
This should skip for private multiplication in order to pass. This can be done by using:

```
# Skipping private multiplication since we expect it to fail
if func in ["mul", "mul_"] and tensor_type == MPCTensor:
    continue
```

We could also assert an error with:

```
with self.assertRaises(NotImplementedError):
     # code here
 ```